apps/frontend/src/modules/surveys/components/studio/create/preview-panel/index.tsx

```diff
@@ -42,7 +43,7 @@ export const Preview = ({ className, mode, viewPort, ...props }: Props) => {
                     ...survey,
                     theme: theme ? { ...theme, colorScheme: JSON.stringify(theme.colorScheme) } : {},
                     questions: parsedQuestions.map((q) => {
-                        if (q.id === question?.id) {
+                        if (q.orderNumber === question?.orderNumber) {
```

Comment:
@AfeezGL Why did you use `orderNumber` here instead of `id`? Remember, this will mutable when we implement reordering survey questions.

Reply:
New questions were not showing after creation because the `id` is temporary until `createQuestionMutation` completes so there's a point where the `question.id` is different from the one from the same question in `parsedQuestions`. 
I used `orderNumber` to find the created question, I have now changed that to `reference`.